src/v3/src/util/theme.ts

```diff
@@ -10,170 +10,126 @@
-          .set('hsl.l', '+0.62')
-          .hex(),
-        dark: chroma(main)
-          .set('hsl.l', '+0.31')
-          .hex(),
-        contrastText: getInverseTextColor(main),
-      };
+    const primaryColor = new Color({
+      name: 'Custom Hue',
+      // Leonardo will throw "Invalid Color Key" error if string is not of type CssColor
+      colorKeys: [main as CssColor],
+      ratios: ODYSSEY_RATIOS,
+    });
+    // SIW always has a white background color
+    const backgroundColor = new BackgroundColor({
+      name: 'Background Color',
+      colorKeys: [WHITE_HEX],
+      ratios: [],
+    });
+    const leonardoTheme = new Theme({
+      colors: [primaryColor],
+      backgroundColor,
+      lightness: 100,
+    }).contrastColorPairs;
+
+    // Remap Leonardo theme to use Odyssey color stops from 50-900
+    const theme = {
+      CustomHue50: leonardoTheme.CustomHue100,
+      CustomHue100: leonardoTheme.CustomHue200,
+      CustomHue200: leonardoTheme.CustomHue300,
+      CustomHue300: leonardoTheme.CustomHue400,
+      CustomHue400: leonardoTheme.CustomHue500,
+      CustomHue500: leonardoTheme.CustomHue600,
+      CustomHue600: leonardoTheme.CustomHue700,
+      CustomHue700: leonardoTheme.CustomHue800,
+      CustomHue800: leonardoTheme.CustomHue900,
+      CustomHue900: leonardoTheme.CustomHue1000,
+    };
+    return theme;
   } catch (err) {
     console.warn(err);
-    return {};
+    return null;
   }
 };
 
-export const createTheme = (
+export const createThemeAndTokens = (
   brandColors?: BrandColors,
   customTokens?: Partial<DesignTokensOverride>,
-): ThemeOptions => {
-  // Odyssey 1.x does not export their theme, but we can recreate it
-  const theme = createOdysseyMuiTheme({ odysseyTokens: { ...Tokens, ...customTokens } });
+): { themeOverride: ThemeOptions, tokensOverride: DesignTokensOverride } => {
+  const tokensOverride = { ...customTokens };
 
-  // TODO: OKTA-667943 - Update palette generation for OD 1.x larger hue range
   if (brandColors?.primaryColor) {
     const p = generatePalette(brandColors.primaryColor);
-    set(theme, 'palette.primary.lighter', p.lighter);
-    set(theme, 'palette.primary.light', p.light);
-    set(theme, 'palette.primary.main', p.main);
-    set(theme, 'palette.primary.dark', p.dark);
-    set(theme, 'palette.primary.contrastText', p.contrastText);
-  }
-  if (customTokens) {
-    if (customTokens.PalettePrimaryMain) {
-      const p = generatePalette(customTokens.PalettePrimaryMain);
-      set(theme, 'palette.primary.lighter', customTokens.PalettePrimaryLighter ?? p.lighter);
-      set(theme, 'palette.primary.light', customTokens.PalettePrimaryLight ?? p.light);
-      set(theme, 'palette.primary.main', customTokens.PalettePrimaryMain ?? p.main);
-      set(theme, 'palette.primary.dark', customTokens.PalettePrimaryDark ?? p.dark);
-      set(theme, 'palette.primary.contrastText', p.contrastText);
-    }
-    if (customTokens.PaletteDangerMain) {
```

Comment:
There was some miscommunication with the pre-1.x implementation of palette generation. We thought we should be generating palettes whenever the customer passed in token overrides for any of the `Main` hues but @alexdahl-okta clarified that palette generation should only be via `brandColors.primaryColor`